See http://lists.gnu.org/archive/html/bug-hurd/2017-10/msg00011.html
coalescing areas with differing objects may not be so good an idea. It notably
potentially coalesces areas not to be swappable (such as IDE write buffers) with
swappable areas.

Index: gnumach/vm/vm_map.c
===================================================================
--- gnumach.orig/vm/vm_map.c
+++ gnumach/vm/vm_map.c
@@ -1033,7 +1033,8 @@ kern_return_t vm_map_enter(
 	 *	extend from below.]
 	 */
 
-	if ((object == VM_OBJECT_NULL) &&
+	if (0 &&
+	    (object == VM_OBJECT_NULL) &&
 	    (entry != vm_map_to_entry(map)) &&
 	    (entry->vme_end == start) &&
 	    (!entry->is_shared) &&
